fw/sock_srv.c

```diff
@@ -708,6 +713,30 @@ tfw_sock_srv_start_srv(TfwSrvGroup *sg, TfwServer *srv, void *hm)
 	return 0;
 }
 
+TfwSrvConn *
+tfw_sock_srv_new_steal_sk(TfwSrvConn *srv_conn)
+{
+	TfwSrvConn *conn;
+	TfwServer *srv = (TfwServer *)srv_conn->peer;
+
+	if (!(conn = tfw_sock_srv_new_conn(srv))) {
```

Comment:
`tfw_sock_srv_new_conn()` -> `tfw_srv_conn_alloc()` -> `might_sleep()` and `kmalloc(GFP_KERNEL)`, so there is a bug. I don't think we need to allocate a new server connection since websockets don't have any server logic, which we have for HTTP connections. What we need to do here is to allocate a new `TfwConn` and probably just need a new memory cache for it.

Reply:
I addressed this by moving connection allocation and stealing into websocket module. And now backend connection is simple connection, but client connection leaved `TfwCliConn`.